### DISCUSSION TITLE
PAG初始化失败，提示permission denied, open ../../utils/libpag.wasm.br

### DISCUSSION THREAD

**Discussion by leung0727** [Tencent/libpag#1737] [Q&A]
小程序开发版本为taro3.4+vue3，在接入PAG小程序包libpag-miniprogram，初始化时提示我没有权限访问libpag.wasm.br文件，导致初始化失败

**Comment by zhalice2011** [comment]
libpag.wasm.br 是根据文档配置的在本地吗?

![wecom-temp-9924351cee2837a2baabc010f213c36e](https://github.com/Tencent/libpag/assets/24289349/9c2fd412-4191-4693-a7ce-6625a4a72abf)

你的 demo 片段可以提供吗

**Comment by DamperHa** [comment]
我也遇到了这个问题，

在初始的时候，和官方文档不同的是，修改了utils的路径。

```
    this.PAG = await PAGInit({
      locateFile: (file) => {
        const filePath = '../../utils/' + file;
        console.log('Trying to load .wasm file from: ', filePath);
        return filePath;
    }});
```
错误信息如下：
```
Trying to load .wasm file from:  ../../utils/libpag.wasm.br
index.js:3309 failed to asynchronously prepare wasm: Error: permission denied, open ../../utils/libpag.wasm.br
(anonymous) @ index.js:3309
Promise.then (async)
instantiateArrayBuffer @ index.js:3308
instantiateAsync @ index.js:3324
createWasm @ index.js:3337
(anonymous) @ index.js:8818
PAGInit @ index.js:9143
_callee3$ @ tmp.js? [sm]:85
s @ regeneratorRuntime.js:1
(anonymous) @ regeneratorRuntime.js:1
(anonymous) @ regeneratorRuntime.js:1
asyncGeneratorStep @ asyncToGenerator.js:1
c @ asyncToGenerator.js:1
(anonymous) @ asyncToGenerator.js:1
(anonymous) @ asyncToGenerator.js:1
onReady @ tmp.js? [sm]:101
index.js:3252 Aborted(Error: permission denied, open ../../utils/libpag.wasm.br)
abort @ index.js:3252
(anonymous) @ index.js:3310
Promise.then (async)
instantiateArrayBuffer @ index.js:3308
instantiateAsync @ index.js:3324
createWasm @ index.js:3337
(anonymous) @ index.js:8818
PAGInit @ index.js:9143
_callee3$ @ tmp.js? [sm]:85
s @ regeneratorRuntime.js:1
(anonymous) @ regeneratorRuntime.js:1
(anonymous) @ regeneratorRuntime.js:1
asyncGeneratorStep @ asyncToGenerator.js:1
c @ asyncToGenerator.js:1
(anonymous) @ asyncToGenerator.js:1
(anonymous) @ asyncToGenerator.js:1
onReady @ tmp.js? [sm]:101
index.js:9150 run time error(env: macOS,mp,1.06.2409140; lib: 3.6.5)
(anonymous) @ index.js:9150
Promise.catch (async)
PAGInit @ index.js:9149
_callee3$ @ tmp.js? [sm]:85
s @ regeneratorRuntime.js:1
(anonymous) @ regeneratorRuntime.js:1
(anonymous) @ regeneratorRuntime.js:1
asyncGeneratorStep @ asyncToGenerator.js:1
c @ asyncToGenerator.js:1
(anonymous) @ asyncToGenerator.js:1
(anonymous) @ asyncToGenerator.js:1
onReady @ tmp.js? [sm]:101
Error: PAGInit fail! Please check .wasm file path valid.
    at index.js:9151(env: macOS,mp,1.06.2409140; lib: 3.6.5)
```

**Reply by DamperHa** [reply]
> 我也遇到了这个问题，
> 
> 在初始的时候，和官方文档不同的是，修改了utils的路径。
> 
> ```
>     this.PAG = await PAGInit({
>       locateFile: (file) => {
>         const filePath = '../../utils/' + file;
>         console.log('Trying to load .wasm file from: ', filePath);
>         return filePath;
>     }});
> ```
> 
> 错误信息如下：
> 
> ```
> Trying to load .wasm file from:  ../../utils/libpag.wasm.br
> index.js:3309 failed to asynchronously prepare wasm: Error: permission denied, open ../../utils/libpag.wasm.br
> (anonymous) @ index.js:3309
> Promise.then (async)
> instantiateArrayBuffer @ index.js:3308
> instantiateAsync @ index.js:3324
> createWasm @ index.js:3337
> (anonymous) @ index.js:8818
> PAGInit @ index.js:9143
> _callee3$ @ tmp.js? [sm]:85
> s @ regeneratorRuntime.js:1
> (anonymous) @ regeneratorRuntime.js:1
> (anonymous) @ regeneratorRuntime.js:1
> asyncGeneratorStep @ asyncToGenerator.js:1
> c @ asyncToGenerator.js:1
> (anonymous) @ asyncToGenerator.js:1
> (anonymous) @ asyncToGenerator.js:1
> onReady @ tmp.js? [sm]:101
> index.js:3252 Aborted(Error: permission denied, open ../../utils/libpag.wasm.br)
> abort @ index.js:3252
> (anonymous) @ index.js:3310
> Promise.then (async)
> instantiateArrayBuffer @ index.js:3308
> instantiateAsync @ index.js:3324
> createWasm @ index.js:3337
> (anonymous) @ index.js:8818
> PAGInit @ index.js:9143
> _callee3$ @ tmp.js? [sm]:85
> s @ regeneratorRuntime.js:1
> (anonymous) @ regeneratorRuntime.js:1
> (anonymous) @ regeneratorRuntime.js:1
> asyncGeneratorStep @ asyncToGenerator.js:1
> c @ asyncToGenerator.js:1
> (anonymous) @ asyncToGenerator.js:1
> (anonymous) @ asyncToGenerator.js:1
> onReady @ tmp.js? [sm]:101
> index.js:9150 run time error(env: macOS,mp,1.06.2409140; lib: 3.6.5)
> (anonymous) @ index.js:9150
> Promise.catch (async)
> PAGInit @ index.js:9149
> _callee3$ @ tmp.js? [sm]:85
> s @ regeneratorRuntime.js:1
> (anonymous) @ regeneratorRuntime.js:1
> (anonymous) @ regeneratorRuntime.js:1
> asyncGeneratorStep @ asyncToGenerator.js:1
> c @ asyncToGenerator.js:1
> (anonymous) @ asyncToGenerator.js:1
> (anonymous) @ asyncToGenerator.js:1
> onReady @ tmp.js? [sm]:101
> Error: PAGInit fail! Please check .wasm file path valid.
>     at index.js:9151(env: macOS,mp,1.06.2409140; lib: 3.6.5)
> ```

已经解决了，将路径修改为绝对路径就好了。'../../utils/'  > "/utils"
